src/Sylius/Behat/Client/ResponseChecker.php

```diff
@@ -132,18 +132,51 @@ public function hasItemWithValue(Response $response, string $key, $value): bool
         return false;
     }
 
-    /** @param string|int $value */
-    public function hasSubResourceWithValue(Response $response, string $subResource, string $key, $value): bool
-    {
-        foreach ($this->getResponseContentValue($response, $subResource) as $resource) {
-            if ($resource[$key] === $value) {
-                return true;
+    public function hasValuesInAnySubresourceObjectCollection(
+        Response $response,
+        string $subResourceName,
+        array $expectedValues,
+    ): bool {
+        $resourceCollection = $this->getResponseContentValue($response, $subResourceName);
+
+        $this->assertIsArray($resourceCollection);
+
+        foreach ($resourceCollection as $resource) {
+            $this->assertIsArray($resource);
+            $this->assertAllExpectedKeysArePresent($expectedValues, $resource);
```

Comment:
Won't this potentially fail if a property is nullable? I don't really see a point to it.